libs/pdi-scanner/ts/src/scanner_client.ts

```diff
@@ -146,9 +146,9 @@ export function createPdiScannerClient() {
     }
   }
 
-  // pdictl only processes one command at a time, so once we send a command, we
-  // can wait on a single response.
-  let pendingResponse: Deferred<PdictlResponse> | undefined;
+  // pdictl queues the commands it receives and only processes one command at a
+  // time, so we track the commands we sent in a queue.
+  const pendingResponseQueue: Array<Deferred<PdictlResponse>> = [];
```

Comment:
This may or may not be useful to you: [`DeferredQueue`](https://github.com/votingworks/vxsuite/blob/d48791757a5ad2e15c5980b2cd7320ec5d6c557e/libs/basics/src/deferred.ts#L41).

Reply:
Ooh, very cool. Just tried out using it, but realized that there's no way to tell if the `DeferredQueue` has any pending promises waiting (which we use in the error case to figure out if we received an error in response to a message or as an event).

This does tempt me to distinguish responses from events in pdictl's JSON format...

Reply:
https://github.com/votingworks/vxsuite/pull/4776

Reply:
Updated to use `DeferredQueue` after making that change. It's very clean now!